sdk/lib/opentelemetry/sdk/trace/samplers/consistent_probability_based.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Trace
+      module Samplers
+        # @api private
+        #
+        # Implements consistent sampling based on a probability.
+        class ConsistentProbabilityBased
+          include ConsistentProbabilityTraceState
+
+          attr_reader :description
+
+          def initialize(probability)
+            # TODO: probability should be 0 if it is < 2**-62
+            @probability = probability
+            @p_floor = (Math.frexp(probability)[1] - 1).abs
+            @p_ceil = @p_floor + 1
+            floor = Math.ldexp(1.0, -@p_floor)
+            ceil = Math.ldexp(1.0, -@p_ceil)
+            @p_ceil_probability = (probability - floor) / (ceil - floor)
+            @description = format('ConsistentProbabilityBased{%.6f}', probability)
+          end
+
+          def ==(other)
+            @description == other.description
+          end
+
+          # @api private
+          #
+          # See {Samplers}.
+          def should_sample?(trace_id:, parent_context:, links:, name:, kind:, attributes:) # rubocop:disable Metrics/MethodLength, Metrics/PerceivedComplexity
+            parent_span_context = OpenTelemetry::Trace.current_span(parent_context).context
+            if !parent_span_context.valid?
+              r = generate_r(trace_id)
+              p = probabilistic_p
+              if p < r
```

Comment:
Should this be `p <= r` ? From the spec: `The span would be sampled because p-value 2 is less than or equal to r-value 5.` If so, we should update everywhere.

Reply:
Thanks! Good catch.